src/test/java/com/ably/kafka/connect/MessageSinkMappingTest.java

```diff
@@ -0,0 +1,99 @@
+     * name on record
+     */
+    @Test
+    void testGetSink_messageNameIsAlwaysSink() {
+        final SinkRecord record = new SinkRecord("sink", 0, Schema.BYTES_SCHEMA, "key".getBytes(), Schema.BYTES_SCHEMA, "value", 0);
+        final Message message = sinkMapping.getMessage(record);
+        assertEquals(message.name, "sink");
+    }
+
+    //we always expect byte arrays
+    @Test
+    void testGetSink_messageConversionFailsWithInvalidKeyType() {
+        final SinkRecord record = new SinkRecord("sink", 0, Schema.BYTES_SCHEMA, "key", Schema.BYTES_SCHEMA, "value", 0);
+        assertThrows(ClassCastException.class, () -> {
+            sinkMapping.getMessage(record);
+        });
+    }
+
+    @Test
+    void testGetSink_messageDataIsTheSameWithRecordValue() {
+        final SinkRecord record = new SinkRecord("sink", 0, Schema.BYTES_SCHEMA, "key".getBytes(), Schema.BYTES_SCHEMA, "value", 0);
+        final Object recordValue = record.value();
+        final Object messageData = sinkMapping.getMessage(record).data;
+        assertEquals(recordValue, messageData);
+    }
+
+    @Test
+    void testGetSink_messageId() {
+        final SinkRecord record = new SinkRecord("sink", 0, Schema.BYTES_SCHEMA, "key".getBytes(), Schema.BYTES_SCHEMA, "value", 0);
+        final String messageId = sinkMapping.getMessage(record).id;
+        assertEquals(messageId, String.format("%d:%d:%d", record.topic().hashCode(), record.kafkaPartition(), record.kafkaOffset()));
+    }
+
+    @Test
+    void testGetSink_messageExtras_sentAndReceivedKeysAreTheSame() {
+        final SinkRecord record = new SinkRecord("sink", 0, Schema.BYTES_SCHEMA, "key".getBytes(), Schema.BYTES_SCHEMA, "value", 0);
+        final MessageExtras messageExtras = sinkMapping.getMessage(record).extras;
+        final JsonUtils.JsonUtilsObject kafkaExtras = createKafkaExtras(record);
+        final JsonObject receivedObject = messageExtras.asJsonObject().get("kafka").getAsJsonObject();
+
+        String receivedKey = receivedObject.get("key").getAsString();
+        String sentKey = kafkaExtras.toJson().get("key").getAsString();
+        assertEquals(receivedKey, sentKey);
+    }
+
+    @Test
+    void testGetSink_messageExtras_recordHeaders() {
+        final SinkRecord record = new SinkRecord("sink", 0, Schema.BYTES_SCHEMA, "key".getBytes(), Schema.BYTES_SCHEMA, "value", 0);
+        final MessageExtras messageExtras = sinkMapping.getMessage(record).extras;
+        final JsonObject receivedObject = messageExtras.asJsonObject().get("kafka").getAsJsonObject();
+
+        JsonUtils.JsonUtilsObject headers = JsonUtils.object();
+        for (Header header : record.headers()) {
+            headers.add(header.key(), header.value());
+        }
+        assertEquals(receivedObject.get("headers").getAsJsonArray(), headers.toJson());
+    }
+
+    //a copy of message extras function
+    private JsonUtils.JsonUtilsObject createKafkaExtras(SinkRecord record) {
```

Comment:
We should avoid code duplication, even for tests :wink: And since this method is an exact copy of a method from the `MessageSinkMapping` it could mean that perhaps this should be a utils function with package scope visibility :thinking:

Reply:
I meant to use this as a reference to compare the tests, I removed it here now https://github.com/ably/kafka-connect-ably/pull/42/commits/bf1eff8aa13627dbd122ae1b7149bfa6b9650d04

Reply:
Some missed test fixes are also here https://github.com/ably/kafka-connect-ably/pull/42/commits/8801cac8db294faab4c01d0804b4305770b1c31d